javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +519,375 @@ object Character {
   @inline def compare(x: scala.Char, y: scala.Char): Int =
     x - y
 
+  // UnicodeBlock
+
+  class Subset protected (name: String) {
+    override final def equals(that: Any): scala.Boolean = super.equals(that)
+    override final def hashCode(): scala.Int = super.hashCode
+    override final def toString(): String = name
+  }
+
+  final class UnicodeBlock private (name: String, private val start: Int, private val end: Int) extends Subset(name)
+
+  object UnicodeBlock {
+    // Initial size from script below
+    private[this] val allBlocks: ArrayList[UnicodeBlock] = new ArrayList[UnicodeBlock](221)
+    private[this] val blocksByNormalizedName = new HashMap[String, UnicodeBlock]()
+
+    private[this] def addNameAliases(properName: String, block: UnicodeBlock,
+        includeJVM: scala.Boolean = true): Unit = {
+      // Add normalized aliases
+      val lower = properName.toLowerCase
+      //   lowercase + spaces
+      blocksByNormalizedName.put(lower, block)
+      //   lowercase + no spaces
+      blocksByNormalizedName.put(lower.replace(" ", ""), block)
+      //   tokens as _
+      if (includeJVM) {
```

Comment:
Instead of this, consider adding: `blockByNormalizedName.put(jvmName.toLower, block)` to `addUnicodeBlock`.